probe/sniff/sniffer.go

```diff
@@ -250,21 +250,30 @@ func (s *Sniffer) Merge(p Packet, rpt report.Report) {
 		return
 	}
 
+	addAdjacency := func(t report.Topology, srcNodeID, dstNodeID string) {
+		srcNode, ok := t.NodeMetadatas[srcNodeID]
+		if !ok {
+			srcNode = report.MakeNodeMetadata()
+		}
+		srcNode.Adjacency = srcNode.Adjacency.Add(dstNodeID)
+		t.NodeMetadatas[srcNodeID] = srcNode
+
+		if _, ok := t.NodeMetadatas[dstNodeID]; !ok {
+			t.NodeMetadatas[dstNodeID] = report.MakeNodeMetadata()
```

Comment:
Correct.  From the probes PoV, its just node generation, and now it generates nodes at both 'ends' of the adjacency.  The local node is always tagged with the origin host id, whereas the remote node remains untagger.  The rendering pipeline line uses this info as a clue that it should treat remote ends as pseudo nodes.  NB for local<->local connections, we get the same connection twice, once for each pid, and end up putting host id in each node.